shared/chat/conversation/messages/wrapper/container.js

```diff
@@ -75,13 +75,19 @@ const mergeProps = (stateProps, dispatchProps) => {
     (previous.type === 'text' || previous.type === 'deleted')
 
   const oldEnough =
-    previous &&
-    previous.timestamp &&
-    message.timestamp &&
+    !!previous &&
+    !!previous.timestamp &&
+    !!message.timestamp &&
     message.timestamp - previous.timestamp > howLongBetweenTimestampsMs
 
+  // Always show a timestamp if the previous message is a concise joined/left message
+  const previousIsJoinedLeft =
+    !!previous && (previous.type === 'systemJoined' || previous.type === 'systemLeft')
```

Comment:
why here too

Reply:
these aren't necessary, but flow complains about them too if I reverse the order of `oldEnough || previousIsJoinedLeft` on L88